charts/omnitruck/templates/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "omnitruck.fullname" . }}
+  labels:
+    {{- include "omnitruck.commonLabels" . | nindent 4 }}
+spec:
+  minReadySeconds: 5
+  progressDeadlineSeconds: 60
+  selector:
+    matchLabels:
+      {{- include "omnitruck.selectorLabels" . | nindent 6 }}
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        {{- include "omnitruck.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "omnitruck.serviceAccountName" . }}
+      terminationGracePeriodSeconds: 60
+      securityContext:
+        fsGroup: 22430
+      {{ if .Values.image -}}
+      containers:
+        - name: {{ include "omnitruck.fullname" . }}
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - sleep
+                  - "10"
+          securityContext:
```

Comment:
This is  a good initial step through. I would move securityContext/livenessProbe/ReadinessProbe settings to value.yaml